src/main/java/org/elasticsearch/action/support/replication/TransportShardReplicationOperationAction.java

```diff
@@ -623,7 +783,7 @@ void performOnReplica(final ReplicationState state, final ShardRouting shard, fi
                 // to wait until they get the new mapping through the cluster
                 // state, which is why we recommend pre-defined mappings for
                 // indices using shadow replicas
-                state.onReplicaSuccess();
+                onReplicaSuccess();
```

Comment:
I change shadow replicas handling. I think it's easier to still keep them here, for the case of a relocating primary for example